pyneuroml/swc/LoadSWC.py

```diff
@@ -0,0 +1,266 @@
+        if parent_id == -1:
+            return None
+        else:
+            parent_node = next((n for n in self.nodes if n.id == parent_id), None)
+            return parent_node
+
+    def get_children(self, node_id):
+        """
+        Get a list of child nodes for a given node.
+
+        Args:
+            node_id (int): The ID of the node for which to get the children.
+
+        Returns:
+            list: A list of SWCNode objects representing the children of the given node.
+
+        Raises:
+            ValueError: If the provided node_id is not found in the graph.
+        """
+        parent_node = next((node for node in self.nodes if node.id == node_id), None)
+        if parent_node is None:
+            raise ValueError(f"Node {node_id} not found or has no children")
+
+        children = [node for node in self.nodes if node.parent_id == node_id]
+        parent_node.children = children
+        return children
+
+    def get_nodes_with_multiple_children(self, type_id=None):
+        """
+        Get a list of child nodes for a given node.
+
+        Args:
+        node_id (int): The ID of the node for which to get the children.
+
+        Returns:
+        list: A list of SWCNode objects representing the children of the given node.
+
+        Raises:
+        ValueError: If the provided node_id is not found in the graph.
+        """
+        nodes = []
+        for node in self.nodes:
+            children = self.get_children(node.id)
+            if len(children) > 1 and (type_id is None or node.type == type_id):
+                nodes.append(node)
+
+        if type_id is not None:
+            print(f"Found {len(nodes)} nodes of type {type_id} with multiple children.")
+
+        return nodes
+
+    def get_nodes_by_type(self, type_id):
+        """
+        Get a list of nodes of a specific type.
+
+        Args:
+            type_id (int): The type ID of the nodes to retrieve.
+
+        Returns:
+            list: A list of SWCNode objects that have the specified type ID.
+        """
+        return [node for node in self.nodes if node.type == type_id]
+
+    def get_branch_points(self, *types):
+        """
+        Get all branch points (nodes with multiple children) of the given types.
+
+        Args:
+           *types (int): One or more node type IDs to filter the branch points by.
+              If no types are provided, all branch points in the graph will be returned.
+
+        Returns:
+            list: A list of SWCNode objects that represent branch points (nodes with
+                  multiple children) of the specified types. If no types are provided,
+                  all branch points in the graph are returned.
+        """
+        nodes = []
+        if not types:
+            nodes = self.get_nodes_with_multiple_children()
+        else:
+            for type_id in types:
+                nodes.extend(self.get_nodes_with_multiple_children(type_id))
+
+        return nodes
```

Comment:
we probably want to return a dict with the key being the type and the value being the list of nodes of that type? 

The use case here is "give me branch points in axons and dendrites" and we'll probably want to treat them differently. If not, the user can always iterate over all the values.